types/index.d.ts

```diff
@@ -19,7 +19,7 @@ export interface MaterialTableProps<RowData extends object> {
   isLoading?: boolean;
   title?: string | React.ReactElement<any>;
   options?: Options;
-  parentChildData?: (row: RowData, rows: RowData[]) => RowData[];
+  parentChildData?: (row: RowData, rows: RowData[]) => RowData;
```

Comment:
Shouldn't it be `=> RowData | undefined` to be able to use `rows.find(a => a.id === row.parentId)` as in the [example](https://material-table.com/#/docs/features/tree-data)?
So find returns RowData if it finds a parent for the row and undefined if not.

Reply:
Yep you're right. Just commited the change.